### PR TITLE
Setup electron builder actions

### LINKED ISSUE #5400: Create GitHub actions or Jenkins to build distributions
### Description

Currently the Lisk Desktop application distributions are build and signed locally. Automate this processes of building and signing using GitHub actions or Jenkins.

Jenkins has a semi automation where we are building the distributions but not signing.

### Acceptance criteria

- Build and sign using GitHub actions or Jenkins
- Should be able to trigger the Job on tag or manually

### Additional Information

https://github.com/marketplace/actions/electron-builder-action, https://github.com/samuelmeuli/action-electron-builder
https://medium.com/@johnjjung/building-an-electron-app-on-github-actions-windows-and-macos-53ab69703f7c
https://dev.to/erikhofer/build-and-publish-a-multi-platform-electron-app-on-github-3lnd
https://www.electronjs.org/docs/latest/tutorial/tutorial-publishing-updating
https://www.codiga.io/blog/notarize-sign-electron-app/

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #5400

### How was it solved?

- Create work flow file for electron-build
- Test and confirm release artifacts are correctly working and are installable on the different distributions.

### How was it tested?

- Check action for results https://github.com/LiskHQ/lisk-desktop/actions/workflows/electron-build.yml

### CODE REVIEW

**Review by oskarleonard — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "dist:mac": "electron-builder --mac",
     "dist:linux": "electron-builder --linux --x64 --publish onTag",
     "pack": "yarn && yarn build && yarn clean:dist && yarn dist",
-    "pack:win": "cmd /c yarn && yarn install:electron:dependencies && yarn clean:build && yarn copy:files && yarn build:prod && yarn build:electron && yarn clean:dist && yarn dist:win",
+    "pack:win": "cmd /c yarn && yarn install:electron:dependencies && yarn copy:files && yarn build:prod && yarn build:electron && yarn dist:win && yarn clean:dist:win",
```
Comment (ikem-legend) [inline package.json:33]: The change here now prevents building an unsigned app on Windows

Comment (ManuGowda) [reply package.json:33]: if you use windows powershell it should work?

Comment (ikem-legend) [reply package.json:33]: It doesn't work with Windows Powershell too. The issue is that the app signing CLI, [smtcl](https://docs.digicert.com/en/digicert-keylocker/tools/command-line-interface/smctl.html) isn't installed on my computer and I shouldn't need to install it if I'm building an unsigned version of the app.

Comment (ManuGowda) [reply package.json:33]: ah ok.. I think now that we have this job configured.. we can easily create tag and delete the tag later for testing on windows..

**Review by eniolam1000752 — COMMENTED** [review]
LGTM, I can't approve because I created the PR 😅

**Review by ikem-legend — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**